### ISSUE TITLE
Creation Graph Sample: Stall when start simulation / reset

### ISSUE THREAD

**Issue by simon-ourmachinery** [OurMachinery/themachinery-public#749]
When starting the simulation for the first time there is a stall as well as when reset.
When published game started the runner takes some time as well at the beginning.

**Comment by simon-ourmachinery** [comment]
based on 2021.8-dev

**Comment by tobias-ourmachinery** [comment]
This stall appears to be related to the UI needing to bake out ttf-fonts for each UI Canvas that has been added to the sample. 

Feels like there's need for some smarter font caching for the in-game UI. Reassigning this to you @raphael-ourmachinery . 

![image](https://user-images.githubusercontent.com/26548681/135084049-bac2cc5e-431d-4487-9943-96d92fe0fcaf.png)

**Comment by raphael-ourmachinery** [comment]
UI Canvas bakes fonts asynchronously in master now, I'm not sure if this change completely solved this issue. I suspect that another reason for this stall is that simulate tab has to bake a font too, maybe we can do this asynchronously like is done for the editor. What do you think @leonardo-ourmachinery?

**Comment by niklas-ourmachinery** [comment]
Seems like we're rebaking the same font multiple times in different contexts. Can't we reuse it so it's baked only once?

**Comment by leonardo-ourmachinery** [comment]
Yes, I think the correct solution would be to compress all the "font-baking" code we have in multiple places right now (runner, simulate_tab, canvas) behind some api in some way and make it sure the call is done asynchronously. But I didn't dig down enough when I tried to unify the runner and simulate_tab code and so I never did it. @raphael-ourmachinery sorry for the late reply, do you want to take a stab at it or should I take care of it?
